src/libstore/local-store.cc

```diff
@@ -1332,7 +1332,7 @@ std::pair<std::filesystem::path, AutoCloseFD> LocalStore::createTempDirInStore()
         /* There is a slight possibility that `tmpDir' gets deleted by
            the GC between createTempDir() and when we acquire a lock on it.
            We'll repeat until 'tmpDir' exists and we've locked it. */
-        tmpDirFn = createTempDir(config->realStoreDir, "tmp");
+        tmpDirFn = createTempDir(std::filesystem::path{config->realStoreDir.to_string()}, "tmp");
```

Comment:
Is this a round trip? I think you can do
```suggestion
        tmpDirFn = createTempDir(config->realStoreDir.get(), "tmp");
```